### PR TITLE
Cb 5753 u tests

### PR DESCRIPTION
**Pull request by devnaumov**

### CODE REVIEW

File: webapp/packages/core-client-activity/src/ClientActivityService.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  it('should set isActive to true when updateActivity is called', () => {
+    clientActivityService.updateActivity();
+    expect(clientActivityService.isActive).toBe(true);
+  });
+
+  it('should reset activity after the timeout period', () => {
+    clientActivityService.updateActivity();
+    expect(clientActivityService.isActive).toBe(true);
+
+    jest.advanceTimersByTime(1000 * 60);
```
Comment (sergeyteleshev) [inline webapp/packages/core-client-activity/src/ClientActivityService.test.ts]: I think it is fine to use `INACTIVE_PERIOD_TIME` const here

File: webapp/packages/core-client-activity/src/ClientActivityService.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    expect(clientActivityService.isActive).toBe(false);
+  });
+
+  it('should clear previous timer if updateActivity is called multiple times', () => {
+    clientActivityService.updateActivity();
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+
+    clientActivityService.updateActivity();
+    expect(clearTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenCalledTimes(2);
```
Comment (sergeyteleshev) [inline webapp/packages/core-client-activity/src/ClientActivityService.test.ts]: would be nice to emulate running time here

File: webapp/packages/core-client-activity/src/ClientActivityService.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  it('should clear timer and reset activity when resetActivity is called', () => {
+    clientActivityService.updateActivity();
+    clientActivityService.resetActivity();
+
+    expect(clientActivityService.isActive).toBe(false);
+    expect(clearTimeout).toHaveBeenCalled();
```
Comment (sergeyteleshev) [inline webapp/packages/core-client-activity/src/ClientActivityService.test.ts]: same here for running time. I suggest to emulate cases like it is actually happening

File: webapp/packages/core-client-activity/src/ClientActivityService.test.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  it('should call onActiveStateChange executor with correct value', () => {
+    const onActiveStateChangeSpy = jest.spyOn(clientActivityService.onActiveStateChange, 'execute');
+
+    clientActivityService.updateActivity();
+    expect(onActiveStateChangeSpy).toHaveBeenCalledWith(true);
+
+    jest.advanceTimersByTime(1000 * 60);
+    expect(onActiveStateChangeSpy).toHaveBeenCalledWith(false);
```
Comment (sergeyteleshev) [inline webapp/packages/core-client-activity/src/ClientActivityService.test.ts]: dont forget to mock Executor also so the test is more moduled. cause if something goes wrong in Executor logic, this test is gonna be fallen as well

File: webapp/packages/core-client-activity/src/ClientActivityService.test.ts (hunk 2 above)
Comment (sergeyteleshev) [reply webapp/packages/core-client-activity/src/ClientActivityService.test.ts]: you can randomly tick in range from `0` to `INACTIVE_PERIOD_TIME - 1`

File: webapp/packages/core-localization/src/LocalizationService.test.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  let service: LocalizationService;
+
+  beforeEach(() => {
+    service = new LocalizationService();
+  });
+
+  it('should initialize with default locale', () => {
+    service.register();
+    expect(service.currentLanguage).toBe(DEFAULT_LOCALE.isoCode);
+    expect(service.supportedLanguages.length).toBeGreaterThan(0);
```
Comment (sergeyteleshev) [inline webapp/packages/core-localization/src/LocalizationService.test.ts:24]: lets check all of the languages so we know for sure nothing is got lost

I mean like copy paste the array of registred locales here and check that everything is fine

File: webapp/packages/core-localization/src/LocalizationService.test.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  it('should set supported languages and default to first supported language', () => {
+    const locales: ILocale[] = [
+      { isoCode: 'de', name: 'German', nativeName: 'Deutsch' },
+      { isoCode: 'es', name: 'Spanish', nativeName: 'Español' },
+    ];
+
+    service.setSupportedLanguages(locales);
+    expect(service.supportedLanguages.length).toBe(2);
+    expect(service.currentLanguage).toBe('de');
+  });
```
Comment (sergeyteleshev) [inline webapp/packages/core-localization/src/LocalizationService.test.ts]: please, split this to 2 different tests:

- should set supported languages
- should set default supported language

File: webapp/packages/core-localization/src/LocalizationService.test.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+  it('should return default locale if the current language is not supported', () => {
+    service.setLanguage('es');
+    expect(service.currentLanguage).toBe(DEFAULT_LOCALE.isoCode);
+  });
+
+  it('should throw error if there are no supported languages', () => {
+    service.supportedLanguages = [];
+    expect(() => service.currentLanguage).toThrowError('No language is available');
```
Comment (sergeyteleshev) [inline webapp/packages/core-localization/src/LocalizationService.test.ts:54]: this seems weird that getter can throw an error. I suggest to add TODO to refactor this logic somewhen in the `LocalizationService.ts`

File: webapp/packages/core-localization/src/LocalizationService.test.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+  it('should replace args in the translation string', () => {
+    const token = 'welcome_message';
+    const translationString = 'Welcome, {arg:name}!';
+    service['localeMap'].set('en', new Map([[token, translationString]]));
+    service.setLanguage('en');
+
+    const translation = service.translate(token, undefined, { name: 'John' });
+    expect(translation).toBe('Welcome, John!');
+  });
```
Comment (sergeyteleshev) [inline webapp/packages/core-localization/src/LocalizationService.test.ts:94]: please also test multiple arguments

File: webapp/packages/core-localization/src/LocalizationService.test.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+});
```
Comment (sergeyteleshev) [inline webapp/packages/core-localization/src/LocalizationService.test.ts:118]: I suggest to test also:

- `load();`
- `isLanguageSupported();`

**Review by sergeyteleshev — CHANGES_REQUESTED** [review]

File: webapp/packages/core-localization/src/LocalizationService.test.ts (hunk 5 above)
Comment (devnaumov) [reply webapp/packages/core-localization/src/LocalizationService.test.ts:24]: Don't understand the need, we only want to check default locale here

**Review by sergeyteleshev — APPROVED** [review]

**Pull request merged**